### DISCUSSION TITLE
Initial parameters for crawl fit

### DISCUSSION THREAD

**Discussion by jtfrench12** [bmcclintock/momentuHMM#88] [Q&A] [answered]
I am trying to use the crawlWrap function to impute some missing locations in my data prior to fitting a HMM and I'm having issues with the optimization. In the elephant example of the vignette I see an argument (initial.state) where the initial values of **a** and **P** can be specified, but the current version of crawlWrap does not appear to use that argument. Is there a new way to input these initial values?

Thanks!

-JF

**Comment by bmcclintock** [comment, marked as answer]
Hi JF,

The 'initial.state' argument was removed from crwMLE in crawl (>= 2.2.1), so it has also been removed from momentuHMM (>=1.4.3) and the vignette examples.  If you haven't already, you should probably update both packages.  Assuming the data are decent, if you're having issues with optimization in crawlWrap it could be due to poor initial values in the 'theta' argument.  If so, I'd suggest trying a range of initial values for 'theta' and using the 'retryFits' argument in crawlWrap.  If the data are relatively poor (e.g. lots of gaps and measurement error) you might also need to use the 'prior' argument.

Cheers,
Brett
